### ISSUE TITLE
Running editSettings() resets the discoverability settings

### ISSUE THREAD

**Issue by mikebull** [not-an-aardvark/snoowrap#290]
When I run the following function to update the description of my subreddit, I notice that my discoverability options re-add me to /r/all.

```
public async updateSidebarContent(updatedContent: string): Promise<void> {
    return this.r(this.subreddit).getSettings().then(settings => {
      this.r(this.subreddit).editSettings({ 
        ...settings,
        description: updatedContent 
      })  
    })    
}
```

After checking the output of `getSettings(...)` I noticed that the only thing other than the description is a property called `default_set`, which seems to be tied to this discoverability option.

I considered adding this as a PR, but after checking the Reddit API docs this property doesn't seem to be listed, although it IS listed as a property in a comment on `getSettings()` in `Subreddit.js`, so I thought I'd double-check here first if anyone knew about this property, if this property is actually the culprit, or whether there is a reason it's not included in the `SubredditSettings` type before sending a PR your way.

**Comment by SpyTec** [comment]
This is probably related to #134. There is a PR for it but it's unfinished #154

**Comment by mikebull** [comment]
Ah yeah, that looks like it should resolve the issue, especially adding the catch-all.

Since this PR looks dead, would you be okay with me forking from master and replicating these changes in a new PR?

**Comment by SpyTec** [comment]
Go for it

**Comment by mikebull** [comment]
Thanks for approving/merging. Out of interest, will you be creating a new release in the near future?

**Comment by SpyTec** [comment]
Since we've only done bugfixes and no features, I'll release a patch version this weekend

**Comment by mikebull** [comment]
Sadly, I've unearthed a handful of other settings that are also being reverted to default, namely free-form reports, allowing multiple images to be selected, and the recommendation settings. I'll create a separate PR to add these to the relevant interface(s).

**Comment by SpyTec** [comment]
Yes, this can be avoided by fixing #134. Your fix only added the ability to add settings that were not part of what snoowrap expects. Interfaces are great, but to prevent resetting settings issue #134 needs to be solved

**Comment by mikebull** [comment]
Is there a particular approach you're happy to lean towards? My JavaScript and TypeScript knowledge isn't great, but from what I can tell an indexer should allow users to add additional properties to the `SubredditSettings` interface. Something along the lines of this:

```
export interface SubredditSettings {
  ...
  spoilers_enabled?: boolean;
  default_set?: boolean;
  [key: string]: any;
}
```

**Comment by SpyTec** [comment]
Types should be strongly typed, adding any sort of catch-all will not work correctly as then using a setting name of `foobar` will not throw errors. Your PR added another option and also made it possible to send in any key you want, which is correct in implementation details since the user can then provide something we aren't covering in the interface, but still be strongly typed in TypeScript.

What should be done here is simply expanding on the `SubredditSettings` interface to cover the settings that are missing

**Comment by mikebull** [comment]
That makes sense. I was a bit worried about adding some in, as they don't seen to be referenced in the official Reddit API documentation, but I'll probably work on getting a more complete interface pushed up at some time this weekend.
